actionpack/test/controller/renderers_test.rb

```diff
@@ -87,4 +101,33 @@ def test_adding_csv_rendering_via_renderers_add
   ensure
     ActionController::Renderers.remove :csv
   end
+
+  def test_raises_missing_render_formatter
+    ActionController::Renderers.add :csv do |value, options|
+      send_data _formatters[:csv].call(value, options), type: Mime[:csv]
+    end
+    exception = assert_raise ActionController::MissingSerializingFormat do
+      get :respond_to_mime, format: 'csv'
+    end
+    assert_match /csv/, exception.message
+    assert_equal Mime[:csv], @response.content_type
+    assert_equal "", @response.body
+  ensure
+    ActionController::Renderers.remove :csv
+  end
+
+  def test_adding_csv_rendering_via_renderers_add_with_formatter
+    ActionController::Renderers.add :csv do |value, options|
+      send_data _formatters[:csv].call(value, options), type: Mime[:csv]
+    end
+    TestController.serializing csv: ->(csv, options) { csv.respond_to?(:to_csv) ? csv.to_csv : csv }
+
+    @request.accept = "text/csv"
+    get :respond_to_mime, format: 'csv'
+    assert_equal Mime[:csv], @response.content_type
+    assert_equal "c,s,v", @response.body
+  ensure
+    ActionController::Renderers.remove :csv
+    TestController._formatters.delete(:csv)
```

Comment:
Consider a more aggressive way to ensure all formatters of a format are removed from all serializers like how all renderers can be removed